File: MP.UnmanagedCode/MP.UnmanagedCode.PowerManagementAPI.VBTests/PowerInformationManagerTest.vb
Imports MP.UnmanagedCode.PowerManagementAPI.Models
Imports Xunit

Public Class PowerInformationManagerTest
    Private ReadOnly _powerInformationManager As PowerInformationManager

    Sub New
        _powerInformationManager = New PowerInformationManager()
    End Sub

    <Fact>
    Sub GetPowerInformation_LastSleepTime()
        Dim lastSleepTime As Double = _powerInformationManager.LastSleepTime
        Xunit.Assert.NotEqual(0, lastSleepTime)
    End Sub

    <Fact>
    Sub GetPowerInformation_LastWakeTime()
        Dim lastWakeTime As Double = _powerInformationManager.LastWakeTime
        Xunit.Assert.NotEqual(0, lastWakeTime)
    End Sub

    <Fact>
    Sub GetPowerInformation_SystemBatteryState()
        Dim result As SystemBatteryState = _powerInformationManager.SystemBatteryState
        Xunit.Assert.True(result.AcOnLine) 'true - the system battery charger is currently operating on external power
    End Sub

    <Fact>
    Sub GetPowerInformation_SystemPowerInformation()
        Dim result As SystemPowerInformation = _powerInformationManager.SystemPowerInformation
        Xunit.Assert.Equal(0.ToString(), result.CoolingMode.ToString()) '0 - system is currently in Active cooling mode
        Xunit.Assert.Equal(0.ToString(), result.Idleness.ToString())    'the current idle level
    End Sub
End Class

File: MP.UnmanagedCode/MP.UnmanagedCode.PowerManagementAPI.Tests/PowerInformationManagerTest.cs
using MP.UnmanagedCode.PowerManagementAPI.Models;
using Xunit;

namespace MP.UnmanagedCode.PowerManagementAPI.Tests
{
    public class PowerInformationManagerTest
    {
        private readonly PowerInformationManager _powerInformationManager;

        public PowerInformationManagerTest()
        {
            _powerInformationManager = new PowerInformationManager();
        }

        [Fact]
        public void GetPowerInformation_LastSleepTime()
        {
            var lastSleepTime = _powerInformationManager.LastSleepTime;

            Assert.NotEqual(0, lastSleepTime);
        }

        [Fact]
        public void GetPowerInformation_LastWakeTime()
        {
            var lastWakeTime = _powerInformationManager.LastWakeTime;

            Assert.NotEqual(0, lastWakeTime);
        }

        [Fact]
        public void GetPowerInformation_SystemBatteryState()
        {
            var systemBatteryState = _powerInformationManager.SystemBatteryState;

            Assert.IsType<SystemBatteryState>(systemBatteryState);
            Assert.True(systemBatteryState.AcOnLine); // true - the system battery charger is currently operating on external power
        }

        [Fact]
        public void GetPowerInformation_SystemPowerInformation()
        {
            var systemPowerInformation = _powerInformationManager.SystemPowerInformation;

            Assert.IsType<SystemPowerInformation>(systemPowerInformation);
            Assert.Equal(0, (int)systemPowerInformation.CoolingMode); // 0 - system is currently in Active cooling mode
            Assert.Equal(0, (int)systemPowerInformation.Idleness);    // the current idle level
        }
    }
}
